Assets/MixedRealityToolkit/_Core/Devices/OpenVR/OpenVRDeviceManager.cs

```diff
@@ -179,35 +181,71 @@ private GenericOpenVRController GetOrAddController(XRNodeState xrNodeState)
-                    detectedController.SetupConfiguration(typeof(ViveWandController));
+                    controllerType = typeof(ViveWandController);
                     break;
                 case SupportedControllerType.ViveKnuckles:
-                    detectedController = new ViveKnucklesController(TrackingState.NotTracked, controllingHand, inputSource);
-                    detectedController.SetupConfiguration(typeof(ViveKnucklesController));
+                    controllerType = typeof(GenericOpenVRController);
                     break;
                 case SupportedControllerType.OculusTouch:
-                    detectedController = new OculusTouchController(TrackingState.NotTracked, controllingHand, inputSource);
-                    detectedController.SetupConfiguration(typeof(OculusTouchController));
+                    controllerType = typeof(OculusTouchController);
                     break;
                 case SupportedControllerType.OculusRemote:
+                    controllerType = typeof(OculusRemoteController);
                     break;
+                default:
+                    Debug.LogError($"Unsupported controller type detected.");
+                    return null;
             }
 
+            var pointers = new List<IMixedRealityPointer>();
+
+            if (MixedRealityManager.HasActiveProfile &&
+                MixedRealityManager.Instance.ActiveProfile.IsInputSystemEnabled &&
+                MixedRealityManager.Instance.ActiveProfile.PointerProfile != null)
+            {
+                for (int i = 0; i < MixedRealityManager.Instance.ActiveProfile.PointerProfile.PointerOptions.Length; i++)
+                {
+                    var pointerProfile = MixedRealityManager.Instance.ActiveProfile.PointerProfile.PointerOptions[i];
+
+                    if (pointerProfile.ControllerType.Type == null ||
+                        pointerProfile.ControllerType == controllerType)
+                    {
+                        if (pointerProfile.Handedness == Handedness.Any ||
+                            pointerProfile.Handedness == Handedness.Both ||
+                            pointerProfile.Handedness == controllingHand)
+                        {
+                            var pointerObject = UnityEngine.Object.Instantiate(pointerProfile.PointerPrefab);
+                            var pointer = pointerObject.GetComponent<IMixedRealityPointer>();
+
+                            if (pointer != null)
+                            {
+                                pointers.Add(pointer);
+                            }
+                        }
+                    }
+                }
+            }
+
+            IMixedRealityInputSource inputSource = InputSystem?.RequestNewGenericInputSource($"{CurrentControllerType} Controller {controllingHand}", pointers.ToArray());
+
+            var detectedController = new GenericOpenVRController(TrackingState.NotTracked, controllingHand, inputSource);
```

Comment:
Concerned by this change, as we are not creating each by their type.
Yes, we have generalized the input, but other things as we move forward will be specific.

Reply:
Whoops, yeah this is a bad change. Sorry.